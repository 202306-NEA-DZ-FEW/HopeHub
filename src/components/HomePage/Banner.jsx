import Image from "next/image";
import { useTranslation } from "next-i18next";
import React from "react";

import BookingButton from "../BookingButton/BookingButton";
import IndexBanner from "../../../public/assets/indexBanner.jpg";

export default function Banner() {
    //Function used for translations
    const { t } = useTranslation("common");

    //Displaying the Homepage Banner
    return (
        <div className='Banner-container top-0 relative dark:brightness-75 -mt-24'>
            <Image
                src={IndexBanner}
                alt='Photo by Taryn Elliott: https://www.pexels.com/photo/couple-with-hands-apart-8204503/'
                width='screen'
                height='screen'
                layout='fixed'
            />

            <div className='flex inset-0 items-center absolute'>
                <h1 className='text-lg ml-6 mb-2 sm:text-4xl sm:mb-32 sm:ml-8 lg:text-6xl lg:mb-48 lg:ml-8 font-poppins font-semibold text-NeutralBlack '>
                    {t("WE ARE HERE TO")}
                </h1>
                <h1 className='text-2xl px-1 mb-2 sm:text-6xl sm:mb-32 lg:text-8xl lg:mb-48 lg:px-3 text-violet-400 font-aclonica font-semibold custom-text-shadow'>
                    {t("HELP")}
                </h1>
            </div>

            {/* Adding the button for booking an appointment */}
            <div className='absolute inset-0 flex items-end justify-end mx-6 md:mx-10 mb-6 sm:m-20 lg:mx-10 lg:mb-32'>
                <BookingButton
                    destination='/booking'
                    buttonText='Book An Appointment'
                />
            </div>
        </div>
    );
}
