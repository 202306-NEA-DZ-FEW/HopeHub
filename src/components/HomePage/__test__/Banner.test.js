import renderer from "react-test-renderer";

import Banner from "../Banner";

it("renders correctly", () => {
    const tree = renderer.create(<Banner />).toJSON();
    expect(tree).toMatchSnapshot();
});
