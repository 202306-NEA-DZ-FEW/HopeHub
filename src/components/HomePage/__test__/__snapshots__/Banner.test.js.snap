// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Banner-container top-0 relative dark:brightness-75 -mt-24"
>
  <img
    alt="Photo by Taryn Elliott: https://www.pexels.com/photo/couple-with-hands-apart-8204503/"
    data-nimg="1"
    decoding="async"
    height={40}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style={
      Object {
        "color": "transparent",
      }
    }
    width={40}
  />
  <div
    className="flex inset-0 items-center absolute"
  >
    <h1
      className="text-lg ml-6 mb-2 sm:text-4xl sm:mb-32 sm:ml-8 lg:text-6xl lg:mb-48 lg:ml-8 font-poppins font-semibold text-NeutralBlack"
      style={
        Object {
          "opacity": 0,
          "transform": " translateY(50px)",
        }
      }
    >
      WE ARE HERE TO
    </h1>
    <h1
      className="text-2xl px-1 mb-2 sm:text-6xl sm:mb-32 lg:text-8xl lg:mb-48 lg:px-3 text-violet-400 font-aclonica font-semibold custom-text-shadow"
      style={
        Object {
          "opacity": 0,
          "transform": " translateY(50px)",
        }
      }
    >
      HELP
    </h1>
  </div>
  <div
    className="absolute inset-0 flex items-end justify-end mx-6 md:mx-10 mb-6 sm:m-20 lg:mx-10 lg:mb-32"
  >
    <div>
      <button
        className="w-28 md:w-72 h-10 rounded-md text-xs sm:text-base lg:text-xl font-poppins font-regular bg-Accent text-NeutralBlack dark:text-NeutralWhite uppercase   dark:bg-Dark_Accent dark:hover:bg-[#3E4E68]  hover:bg-[#879AB8] hover:text-NeutralWhite hover:scale-105 duration-500"
        data-testid="booking-button"
      >
        <a
          href="/Auth"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Book An Appointment
        </a>
      </button>
    </div>
  </div>
</div>
`;
